lib/paperclip/io_adapters/abstract_adapter.rb

```diff
@@ -8,8 +8,19 @@ class AbstractAdapter
     delegate :binmode, :binmode?, :close, :close!, :closed?, :eof?, :path, :rewind, :unlink, :to => :@tempfile
     alias :length :size
 
+    def initialize(target, options = {})
+      @target  = target
+      @options = options
+    end
+
     def fingerprint
-      @fingerprint ||= Digest::MD5.file(path).to_s
+      @fingerprint ||= OpenSSL::Digest
+        .new(@options.fetch(:hash_digest, "SHA1")).tap do |digest|
+        File.open(path, 'rb') do |f|
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.